backend/src/order.ts

```diff
@@ -332,34 +332,37 @@ async function getPaymentHandler(order_id: number, provider?: string): Promise<I
 
 export async function preparePayment(order_id: number): Promise<any> {
   let conn: IConnection;
+  let provider = DEFAULT_PROVIDER;
+  let status = 'payment-pending';
 
   log.info('Preparing payment', {order_id: order_id});
   try {
     conn = await db.beginTransaction();
-  } catch (err) {
-    log.error('Failed to prepare payment', {error: err, order_id});
-    throw err;
-  }
 
-  try {
     log.log('debug', 'Get order payment status', {order_id});
-    const res = await db.query('select status from nk2_orders where id = :order_id', {order_id}, conn);
+    const res = await db.query('select status, price as order_price from nk2_orders where id = :order_id', {order_id}, conn);
     
     log.log('debug', 'Reject if status is not seats-reserved', {order_id});
     if (res[0]['status'] !== 'seats-reserved') {
-      const err = {name: "Validation error", message: "Invalid order status"}
-      throw err;
+      throw {name: "Status error", message: "Invalid order status"};
     }
 
-    log.log('debug', 'Set order status to payment-pendig and add payment provider', {order_id});
-    await db.query('update nk2_orders set status = "payment-pending", payment_provider = :provider where id = :order_id', {order_id, provider: DEFAULT_PROVIDER}, conn);
+    log.log('debug', 'Use no-provider as payment provider if order price is under min payment', {order_id});
+    if (res[0]['order_price'] <= MIN_PAYMENT) {
+      log.log('debug', 'Using no-provider as payment provider', {order_id});
+      provider = 'no-provider';
+      status = 'paid';
+    } 
+
+    log.log('debug', 'Set order status to payment-pending and add payment provider', {order_id});
+    await db.query('update nk2_orders set status = :status, payment_provider = :provider where id = :order_id', {order_id, provider, status}, conn);
```

Comment:
If we set the status to "paid" here, do we ever send out the tickets? I think it could work to here use "payment-pending", and then the user gets redirected to "/#ok" which triggers paymentDone() with success verification that'd return always true (or maybe check some hash if you can think of one), and then marks the order as paid and sends tickets. Would keep the flow as similar as possible.

Reply:
If you decide to keep this, maybe change the log line to write the actual status too.

Reply:
Excelent notification. Updated flow:

Create payment -> redirect to success callback -> automatically verified -> redirected to /#ok

Reply:
And also added simple signature for verification in no-provider, might not be neccessary really, but does not hurt anybody